staging/src/k8s.io/apiserver/pkg/util/webhook/webhook.go

```diff
@@ -131,20 +131,25 @@ func WithExponentialBackoff(ctx context.Context, initialBackoff time.Duration, w
 		Steps:    5,
 	}
 
-	var err error
-	wait.ExponentialBackoff(backoff, func() (bool, error) {
-		err = webhookFn()
-		if ctx.Err() != nil {
-			// we timed out or were cancelled, we should not retry
-			return true, err
-		}
-		if shouldRetry(err) {
+	var webhookErr error
```

Comment:
maybe a comment for future generations

`// having a webhook error allows us to track the last actual webhook error for requests that are later cancelled or time out`